From b1a0aef9978d4a41e7e601c277d4fb1b0cfbed89 Mon Sep 17 00:00:00 2001
From: Dave Jones <davej@codemonkey.org.uk>
Date: Thu, 26 Aug 2021 11:17:34 -0400
Subject: [PATCH] Use fcntl.h for dev_t & mode_t
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Should fix:

In file included from /usr/include/bits/statx.h:31,
                 from /usr/include/sys/stat.h:446,
                 from include/files.h:3,
                 from include/shm.h:6,
                 from syscalls/x86/modify_ldt.c:12:
/usr/include/linux/stat.h:57:2: error: unknown type name ‘__s64’
  __s64 tv_sec;
  ^~~~~
/usr/include/linux/stat.h:58:2: error: unknown type name ‘__u32’
  __u32 tv_nsec;
  ^~~~~
/usr/include/linux/stat.h:59:2: error: unknown type name ‘__s32’
  __s32 __reserved;
  ^~~~~
/usr/include/linux/stat.h:101:2: error: unknown type name ‘__u32’
  __u32 stx_mask; /* What results were written [uncond] */

[Retrieved from:
https://github.com/kernelslacker/trinity/commit/b1a0aef9978d4a41e7e601c277d4fb1b0cfbed89]
Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
---
 include/files.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/files.h b/include/files.h
index 2a8e0e2d..ba3cc98f 100644
--- a/include/files.h
+++ b/include/files.h
@@ -1,6 +1,6 @@
 #pragma once
 
-#include <sys/stat.h>
+#include <fcntl.h>
 #include "fd.h"
 
 unsigned long get_o_flags(void);
